### PR TITLE
Serialize Page Graph args as json.

### PR DESCRIPTION
**Pull request by goodov**
#### Better handling of WebAPI values

WebAPI values are now serialized more effectively. It includes all the
properties of objects, even the ones that are not easily visible. This makes it
similar to how DevTools handles object properties.

The `EdgeJSCall::args` is now a JSON array that contains all the observed
arguments. The `EdgeJSResult::result` is also a JSON value.

#### Improved tracking of Console API

The `Console` built-in functionality in V8 is now being tracked. This means that
all `console.*` APIs will be visible in the graph as JSBuiltin `ConsoleLog`,
`ConsoleWarn`, and similar.

The `console.log` WebAPI has been renamed to `ConsoleMessageAdded` to better
represent its tracking functionality. However, it may not appear in most graphs
as it is usually not bound to an acting script, and such calls are currently
ignored in the graph.

#### Expanded tracking of APIs

The `Geolocation` and `MediaDevices` WebAPIs are now being tracked.

Resolves https://github.com/brave/brave-browser/issues/35130

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by pes10k** [comment]
This all looks wonderful @goodov ! I saw some canvas and webgl specific changes/files were removed.  Was just curious, is that bc the new change generalizes what were special cases before, or did something else related to those systems change?

**Review by pes10k — APPROVED** [review]

**Comment by goodov** [comment]
> I saw some canvas and webgl specific changes/files were removed. Was just curious, is that bc the new change generalizes what were special cases before, or did something else related to those systems change?

yep, the new approach covers all these things automatically so the manual serialization of these structs were removed.

**Comment by goodov** [comment]
@pes10k @L3thal14 would be nice if you could poke these builds and see if everything's good. These are built on this exact branch:

(just choose the one you're most comfortable with)
https://build-artifacts.brave.com/test-brave-browser-build-linux-x64/992/brave-browser-nightly-1.65.77-linux-amd64.zip
https://build-artifacts.brave.com/test-brave-browser-build-macos-arm64/553/brave-v1.65.77-darwin-arm64.zip
https://build-artifacts.brave.com/test-brave-browser-build-macos-x64/1168/brave-v1.65.77-darwin-x64.zip
https://build-artifacts.brave.com/test-brave-browser-build-windows-x64/1274/brave-v1.65.77-win32-x64.zip

**Comment by L3thal14** [comment]
> @pes10k @L3thal14 would be nice if you could poke these builds and see if everything's good. These are built on this exact branch:
> 
> (just choose the one you're most comfortable with) https://build-artifacts.brave.com/test-brave-browser-build-linux-x64/992/brave-browser-nightly-1.65.77-linux-amd64.zip https://build-artifacts.brave.com/test-brave-browser-build-macos-arm64/553/brave-v1.65.77-darwin-arm64.zip https://build-artifacts.brave.com/test-brave-browser-build-macos-x64/1168/brave-v1.65.77-darwin-x64.zip https://build-artifacts.brave.com/test-brave-browser-build-windows-x64/1274/brave-v1.65.77-win32-x64.zip

I tried the `linux-amd64` build (https://geolocation-pp-testing.netlify.app/). The `Geolocation` and `MediaDevices` WebAPIs are being appropriately tracked with args:
```
{'node type': 'web API', 'id': '245', 'timestamp': '2367', 'method': 'MediaDevices.getUserMedia'}
{'edge type': 'js call', 'id': '246', 'timestamp': '2367', 'args': '[{"audio":1,"video":0}]', 'script position': 2195}


{'node type': 'web API', 'id': '236', 'timestamp': '2366', 'method': 'Geolocation.getCurrentPosition'}
{'edge type': 'js call', 'id': '237', 'timestamp': '2366', 'args': '["function success(position) {\\n                const latitude = position.coords.latitude;\\n                const longitude = position.coords.longitude;\\n\\n                status.textContent = \\"\\";\\n                mapLink.href = `https://www.openstreetmap.org/#map=18/${latitude}/${longitude}`;\\n                mapLink.textContent = `Latitude: ${latitude} °, Longitude: ${longitude} °`;\\n                const url = `https://example.com/?longitude=${longitude}&latitude=${latitude}`;\\n                fetch(url)\\n                .then(response => {\\n                    if (!response.ok) {\\n                    throw new Error(`Network response was not ok: ${response.status}`);\\n                    }\\n                    return response.json();\\n                })\\n                .then(data => {\\n                    // Handle the response data here\\n                    console.log(data);\\n                })\\n                .catch(error => {\\n                    console.error(\'Fetch error:\', error);\\n                });\\n            }","function errorCallBack(error) {\\n                status.textContent = \\"Unable to retrieve your location\\";\\n                console.log(`ERROR(${error.code}): ${error.message}`);\\n            }",{"enableHighAccuracy":0,"maximumAge":0,"timeout":4294967295.0}]', 'script position': 1672}

```
But, the result JSON seems to return `Symbol(Symbol.toStringTag)` for most cases because of `JSON.stringify` call I guess? I would expect a [GeolocationPosition](https://developer.mozilla.org/en-US/docs/Web/API/GeolocationPosition) object in JSON format for `Geolocation` :
```
{'node type': 'web API', 'id': '228', 'timestamp': '2366', 'method': 'Navigator.geolocation.get'}
{'edge type': 'js result', 'id': '230', 'timestamp': '2366', 'value': '{"Symbol(Symbol.toStringTag)":"Geolocation"}'}

{'node type': 'web API', 'id': '238', 'timestamp': '2367', 'method': 'Navigator.mediaDevices.get'}
{'edge type': 'js result', 'id': '240', 'timestamp': '2367', 'value': '{"Symbol(Symbol.toStringTag)":"MediaDevices","ondevicechange":null}'}
```
The console APIs are being properly tracked as well.

**Comment by pes10k** [comment]
@goodov this looks great to me

@L3thal14 `JSON.stringify(navigator.geolocation)` returns `{}` though, so this PR seems correct (even if there might be a bug or other unexpected reason thats what `JSON.stringify` returns here. So, i think the PR is correct, and best I can tell, this is ready to merge and go

**Comment by goodov** [comment]
> `Symbol(Symbol.toStringTag)`

This comes from the devtools serializer:
image

I can remove it, but decided to keep it in case it might be usable somewhere.

> `JSON.stringify`

We can't use `JSON.stringify`, because it doesn't serialize non-enumerable properties, which you actually want most of the time while working on PG-generated graphs.

File: chromium_src/v8/src/inspector/value-mirror.cc [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    return true;
+  }
+
+  auto& mirrors() { return m_mirrors_; }
+
+ private:
+  std::vector<v8_inspector::PropertyMirror> m_mirrors_;
+};
+
+// Returns true if the value is an empty object.
```
Comment (iefremov) [inline chromium_src/v8/src/inspector/value-mirror.cc:34]: Frankly, there are some other places which deserve a comment much more than this function :D

File: chromium_src/v8/src/inspector/value-mirror.cc [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+#if BUILDFLAG(ENABLE_BRAVE_PAGE_GRAPH)
+
+#include "brave/v8/include/v8-isolate-page-graph-utils.h"
+#include "v8-local-handle.h"
+#include "v8-primitive.h"
+
+namespace v8::page_graph {
+
+namespace {
+
```
Comment (iefremov) [inline chromium_src/v8/src/inspector/value-mirror.cc:19]: file-level comment please

**Review by iefremov — APPROVED** [review]

**Pull request merged**